xapian-core/tests/harness/backendmanager.cc

```diff
@@ -145,14 +155,36 @@ BackendManager::get_database(const std::string &dbname,
     tmp_path += '~';
 
     {
-	Xapian::WritableDatabase wdb = get_writable_database(tmp_dbleaf,
-							     string());
+	Xapian::WritableDatabase wdb;
+	if (is_remote_bm) {
+	    wdb = get_writable_database_from_sub_manager(tmp_dbleaf);
+	} else if (is_honey_bm) {
+	    wdb = get_alt_writable_database(tmp_dbleaf, string());
+	} else {
+	    wdb = get_writable_database(tmp_dbleaf, string());
+	}
```

Comment:
I think it would be better to implement this dispatch via the polymorphism of the `BackendManager` class hierarchy instead of setting boolean flags based on string operations on the backend name.

The essence of what we're wanting to do here is "get a WritableDatabase for generating" so I'd suggest a new virtual method called say `get_generated_database()`, and then each of these alternatives forms the body of the implementation of that for the respective `BackendManager` subclass, e.g.:

```
Xapian::WritableDatabase
BackendManagerRemote::get_generated_database(const string& path)
{
    return get_writable_database_from_sub_manager(path);
}
```

The `else` case would then become the default implementation of this in the base class.

This also means this backend-specific code is in the appropriate `BackendManager` subclass, which is what we ideally want.

Reply:
Added new function.